### PR TITLE
Add metadata to get, fget and get_partial methods

### PR DESCRIPTION
**Pull request by DrJackilD**
Regard to this issue: https://github.com/minio/minio-py/issues/521 I want to add support for additional headers in `get` methods.

### CODE REVIEW

File: minio/api.py [hunk 1]
```diff
@@ -698,6 +698,8 @@ def fget_object(self, bucket_name, object_name, file_path):
         :param bucket_name: Bucket to read object from.
         :param object_name: Name of the object to read.
         :param file_path: Local file path to save the object.
+        :param metadata: Any additional metadata to be uploaded along
```
Comment (harshavardhana) [inline minio/api.py]: Are you sure this is what you want to do? GET doesn't upload any object in-fact it downloads the object. How does adding another parameter to fget_object help here? .

Comment (DrJackilD) [inline minio/api.py]: As example I can provide a case, when you using SSE-C and want to provide custom headers to get decrypted file from S3:

```python
    key = b'32byteslongsecretkeymustprovided'
    encryption_key = base64.b64encode(key).decode()
    encryption_key_md5 = base64.b64encode(hashlib.md5(key).digest()).decode()

    minio = Minio('s3.amazonaws.com', access_key=AWSAccessKeyId, secret_key=AWSSecretKey)

    obj = minio.get_object('minio-vfs-test', 'test_crypt.txt', metadata={
        'x-amz-server-side-encryption-customer-algorithm': 'AES256',
        'x-amz-server-side-encryption-customer-key': encryption_key,
        'x-amz-server-side-encryption-customer-key-MD5': encryption_key_md5
    })

    print(obj.read())

    minio.put_object('minio-vfs-test', 'test_crypt1.txt', content, content.getbuffer().nbytes,
                     metadata={
                         'x-amz-server-side-encryption-customer-algorithm': 'AES256',
                         'x-amz-server-side-encryption-customer-key': encryption_key,
                         'x-amz-server-side-encryption-customer-key-MD5': encryption_key_md5
                     })

    obj = minio.get_object('minio-vfs-test', 'test_crypt1.txt', metadata={
        'x-amz-server-side-encryption-customer-algorithm': 'AES256',
        'x-amz-server-side-encryption-customer-key': encryption_key,
        'x-amz-server-side-encryption-customer-key-MD5': encryption_key_md5
    })

    print(obj.read())
```

**Comment by harshavardhana** [comment]
> key = b'32byteslongsecretkeymustprovided'
>     encryption_key = base64.b64encode(key).decode()
>     encryption_key_md5 = base64.b64encode(hashlib.md5(key).digest()).decode()
> 
>     minio = Minio('s3.amazonaws.com', access_key=AWSAccessKeyId, secret_key=AWSSecretKey)
> 
>     obj = minio.get_object('minio-vfs-test', 'test_crypt.txt', metadata={
>         'x-amz-server-side-encryption-customer-algorithm': 'AES256',
>         'x-amz-server-side-encryption-customer-key': encryption_key,
>         'x-amz-server-side-encryption-customer-key-MD5': encryption_key_md5
>     })
> 
>     print(obj.read())
> 
>     minio.put_object('minio-vfs-test', 'test_crypt1.txt', content, content.getbuffer().nbytes,
>                      metadata={
>                          'x-amz-server-side-encryption-customer-algorithm': 'AES256',
>                          'x-amz-server-side-encryption-customer-key': encryption_key,
>                          'x-amz-server-side-encryption-customer-key-MD5': encryption_key_md5
>                      })
> 
>     obj = minio.get_object('minio-vfs-test', 'test_crypt1.txt', metadata={
>         'x-amz-server-side-encryption-customer-algorithm': 'AES256',
>         'x-amz-server-side-encryption-customer-key': encryption_key,
>         'x-amz-server-side-encryption-customer-key-MD5': encryption_key_md5
>     })
> 
>     print(obj.read())

Then perhaps you need to update the comment.. to reflect the right purpose here.  Rather than saying `upload` we should say that these are request headers.. and not `metadata` but rather just `request_headers` . Since these are request_headers http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectGET.html#get-object-sse-c-specific-request-headers

With this we can also support now 'If-Match', 'If-Modified-Since' etc. 

Additionally it would be cleaner to get a full example added as well to `examples/`

**Comment by DrJackilD** [comment]
@harshavardhana I'm agreed, that `request_headers` will be cleaner, I'm just take same name, as in the `put_object`, where this variable in fact just additional headers too. So, I suggest to left it as is, to be consistent with other methods. Agreed?

P. S. Also I saw that CI was failed, I'm looking what's happening now

**Comment by DrJackilD** [comment]
Tests fixed and documentation is updated with example of usage

File: minio/api.py [hunk 2]
```diff
@@ -698,8 +698,7 @@ def fget_object(self, bucket_name, object_name, file_path, metadata=None):
         :param bucket_name: Bucket to read object from.
         :param object_name: Name of the object to read.
         :param file_path: Local file path to save the object.
-        :param metadata: Any additional metadata to be uploaded along
-            with your GET request.
+        :param metadata: Any additional metadata to be added to GET request's headers.
```
Comment (harshavardhana) [inline minio/api.py]: IMHO we should keep it as request headers,  in put_object it is meant to be saved with the object. Here it is meant to be queried.. so it is better to named request_headers or req_headers makes the code more readable.

Comment (DrJackilD) [inline minio/api.py]: Ok, agreed, I will rename it right now :)

**Comment by DrJackilD** [comment]
So, as far as you approved this PR, when I could expect this on the master branch? I want to use it in my project :)

**Pull request merged**